### ISSUE TITLE
Occasionally breaks pip installs of other software

### ISSUE THREAD

**Issue by cathalgarvey** [fafhrd91/setuptools-rust#4]
I just got this error:

```bash
cathal@europa:~/Projects/python/PyGram/src$ sudo pip2 install ipython
[sudo] password for cathal: 
The directory '/home/cathal/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/cathal/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting ipython
  Downloading ipython-5.3.0-py2-none-any.whl (750kB)
    100% |████████████████████████████████| 757kB 420kB/s 
Requirement already satisfied (use --upgrade to upgrade): setuptools>=18.5 in /usr/lib/python2.7/dist-packages (from ipython)
Collecting backports.shutil-get-terminal-size; python_version == "2.7" (from ipython)
  Downloading backports.shutil_get_terminal_size-1.0.0-py2.py3-none-any.whl
Collecting pickleshare (from ipython)
  Downloading pickleshare-0.7.4-py2.py3-none-any.whl
Collecting simplegeneric>0.8 (from ipython)
  Downloading simplegeneric-0.8.1.zip
Collecting decorator (from ipython)
  Downloading decorator-4.0.11-py2.py3-none-any.whl
Collecting prompt-toolkit<2.0.0,>=1.0.4 (from ipython)
  Downloading prompt_toolkit-1.0.14-py2-none-any.whl (248kB)
    100% |████████████████████████████████| 256kB 668kB/s 
Collecting pexpect; sys_platform != "win32" (from ipython)
  Downloading pexpect-4.2.1-py2.py3-none-any.whl (55kB)
    100% |████████████████████████████████| 61kB 871kB/s 
Collecting pathlib2; python_version == "2.7" or python_version == "3.3" (from ipython)
  Downloading pathlib2-2.2.1-py2.py3-none-any.whl
Collecting pygments (from ipython)
  Downloading Pygments-2.2.0-py2.py3-none-any.whl (841kB)
    100% |████████████████████████████████| 849kB 534kB/s 
Collecting traitlets>=4.2 (from ipython)
  Downloading traitlets-4.3.2-py2.py3-none-any.whl (74kB)
    100% |████████████████████████████████| 81kB 882kB/s 
Requirement already satisfied (use --upgrade to upgrade): six>=1.9.0 in /usr/lib/python2.7/dist-packages (from prompt-toolkit<2.0.0,>=1.0.4->ipython)
Collecting wcwidth (from prompt-toolkit<2.0.0,>=1.0.4->ipython)
  Downloading wcwidth-0.1.7-py2.py3-none-any.whl
Collecting ptyprocess>=0.5 (from pexpect; sys_platform != "win32"->ipython)
  Downloading ptyprocess-0.5.1-py2.py3-none-any.whl
Collecting scandir; python_version < "3.5" (from pathlib2; python_version == "2.7" or python_version == "3.3"->ipython)
  Downloading scandir-1.5.tar.gz
    Complete output from command python setup.py egg_info:
    running egg_info
    creating pip-egg-info/scandir.egg-info
    writing pip-egg-info/scandir.egg-info/PKG-INFO
    writing top-level names to pip-egg-info/scandir.egg-info/top_level.txt
    writing dependency_links to pip-egg-info/scandir.egg-info/dependency_links.txt
    writing manifest file 'pip-egg-info/scandir.egg-info/SOURCES.txt'
    warning: manifest_maker: standard file '-c' not found
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ex6U0I/scandir/setup.py", line 54, in <module>
        'Programming Language :: Python :: Implementation :: CPython',
      File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/lib/python2.7/dist-packages/setuptools/command/egg_info.py", line 195, in run
        self.find_sources()
      File "/usr/lib/python2.7/dist-packages/setuptools/command/egg_info.py", line 222, in find_sources
        mm.run()
      File "/usr/lib/python2.7/dist-packages/setuptools/command/egg_info.py", line 306, in run
        self.add_defaults()
      File "/usr/lib/python2.7/dist-packages/setuptools/command/egg_info.py", line 335, in add_defaults
        sdist.add_defaults(self)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/sdist.py", line 171, in add_defaults
        build_ext = self.get_finalized_command('build_ext')
      File "/usr/lib/python2.7/distutils/cmd.py", line 312, in get_finalized_command
        cmd_obj.ensure_finalized()
      File "/usr/lib/python2.7/distutils/cmd.py", line 109, in ensure_finalized
        self.finalize_options()
      File "/usr/lib/python2.7/dist-packages/setuptools/command/build_ext.py", line 134, in finalize_options
        self.check_extensions_list(self.extensions)
      File "/usr/local/lib/python2.7/dist-packages/setuptools_rust/build_ext.py", line 16, in check_extensions_list
        _build_ext.check_extensions_list(extensions)
    TypeError: unbound method check_extensions_list() must be called with build_ext instance as first argument (got list instance instead)
```

**Comment by fafhrd91** [comment]
should be fixed in master

**Comment by cathalgarvey** [comment]
Thank you! :)

On 11 April 2017 20:41:31 GMT+01:00, Nikolay Kim <notifications@github.com> wrote:
>Closed #4.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/fafhrd91/setuptools-rust/issues/4#event-1038984097

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
